packages/stream_chat_flutter/lib/src/message_input/audio_message_send_button/stream_audio_message_controllers.dart

```diff
@@ -0,0 +1,186 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:stream_chat_flutter/src/message_input/audio_message_send_button/stream_audio_message_gesture_state.dart';
+import 'package:stream_chat_flutter/stream_chat_flutter.dart';
+
+/// Widget that displays the audio recording controls.
+class StreamAudioMessageControllers extends StatefulWidget {
+  /// Creates a new StreamAudioMessageControllers
+  const StreamAudioMessageControllers({super.key});
+
+  @override
+  State<StreamAudioMessageControllers> createState() =>
+      _StreamAudioMessageControllersState();
+}
+
+class _StreamAudioMessageControllersState
+    extends State<StreamAudioMessageControllers> with TickerProviderStateMixin {
+  Color? iconColor;
+  DateTime? _startTime;
+  Timer? _timer;
+  late final AnimationController _controller;
+
+  @override
+  void didChangeDependencies() {
```

Comment:
Apart from the theme, why are we initialising in didChangeDeps for controllers and such? Any situations where initState was failing?

Reply:
There was an exception thrown when disposing the `_controller`, than indicated to put it in `didChangeDependencies`